File: src/modules/company/services/company_update_status_service.py
from src.shared.types.user_request import UserRequest
from src.modules.company.models import Company
from src.modules.company.domain.dto.company_update_status_dto import (
    CompanyUpdateStatusDTO,
)
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_company_validator import (
    user_company_validator,
)


class CompanyUpdateStatusService:
    @classmethod
    def execute(self, id: int, dto: CompanyUpdateStatusDTO, user_request: UserRequest):

        company = Company.objects.filter(id=id, deleted_at=None).first()

        if company is None:
            raise NotFoundException("Empresa não encontrada!")

        user_company_validator(id, user_request)

        company.status = dto.status

        company.save()

        return company

File: src/modules/company/services/company-update-status.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { UserRequest } from 'src/shared/types/user-request';
import { userCompanyValidator } from 'src/shared/validators/user-company.validator';
import { CompanyUpdateStatusDto } from '../domain/dto/company-update-status.dto';
import { CompanyEntity } from '../domain/entities/company.entity';

@Injectable()
export class CompanyUpdateStatusService {
  constructor(private prisma: PrismaService) {}
  async execute(
    id: number,
    dto: CompanyUpdateStatusDto,
    userRequest: UserRequest,
  ): Promise<CompanyEntity> {
    const company = await this.prisma.company.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
    });

    if (!company) throw new NotFoundException('Empresa não encontrada!');

    await userCompanyValidator(id, userRequest);

    const updatedCompany = await this.prisma.company.update({
      data: {
        status: dto.status,
      },
      where: {
        id: id,
      },
    });

    return new CompanyEntity(updatedCompany);
  }
}
